### PR TITLE
chore: only generate breakpoint mapping on breakpoint variants

### PR DESCRIPTION
**Pull request by felipechiave**
*Issue #, if available:*

*Description of changes:*
Components should only have default breakpoint mapping and transition styles when there is a 'breakpoint' variant

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-react/lib/react-studio-template-renderer.ts [hunk 1]
```diff
@@ -595,7 +597,7 @@ export abstract class ReactStudioTemplateRenderer extends StudioTemplateRenderer
       factory.createBindingElement(
         factory.createToken(ts.SyntaxKind.DotDotDotToken),
         undefined,
-        factory.createIdentifier(hasVariant ? 'restProp' : 'rest'),
+        factory.createIdentifier(hasBreakpoint ? 'restProp' : 'rest'),
```
Comment (SwaySway) [inline packages/codegen-ui-react/lib/react-studio-template-renderer.ts:600]: Are restProps only used for breakpoints?

File: packages/codegen-ui-react/lib/react-studio-template-renderer.ts [hunk 2]
```diff
@@ -837,6 +843,42 @@ export abstract class ReactStudioTemplateRenderer extends StudioTemplateRenderer
     );
   }
 
+  /**
```
Comment (SwaySway) [inline packages/codegen-ui-react/lib/react-studio-template-renderer.ts:846]: not sure I'm seeing the difference between this and `buildOverridesFromVariantsAndRestProp`

File: packages/codegen-ui-react/lib/react-studio-template-renderer.ts (hunk 1 above)
Comment (felipechiave) [reply packages/codegen-ui-react/lib/react-studio-template-renderer.ts:600]: Yes, restProp is used to include a transition style onto components but it should only be added on components with a 'breakpoint' variant.

File: packages/codegen-ui-react/lib/react-studio-template-renderer.ts (hunk 2 above)
Comment (felipechiave) [reply packages/codegen-ui-react/lib/react-studio-template-renderer.ts:846]: `buildOverridesFromVariantsAndRestProp` is used when a 'breakpoint' variant exists on a component as it includes the default breakpoint mapping as an argument into `getOverridesFromVariants`. This function builds the call without that mapping for components that do not have a 'breakpoint' variant.

**Review by SwaySway — APPROVED** [review]

File: packages/codegen-ui-react/lib/react-studio-template-renderer.ts (hunk 2 above)
Comment (felipechiave) [reply packages/codegen-ui-react/lib/react-studio-template-renderer.ts]: Merged functions while conditioning on `hasBreakpoints`

**Review by hein-j — APPROVED** [review]

**Pull request merged**